index/scorch/snapshot_vector_index.go

```diff
@@ -42,16 +43,7 @@ func (is *IndexSnapshot) VectorReader(ctx context.Context, vector []float32,
 		rv.iterators = make([]segment_api.VecPostingsIterator, len(is.segment))
 	}
 
-	for i, seg := range is.segment {
-		if sv, ok := seg.segment.(segment_api.VectorSegment); ok {
-			pl, err := sv.SimilarVectors(field, vector, k, seg.deleted)
-			if err != nil {
-				return nil, err
-			}
-			rv.postings[i] = pl
-			rv.iterators[i] = pl.Iterator(rv.iterators[i])
-		}
-	}
+	// Do this as part of optimizations' Finish() func.
```

Comment:
i think its better to leave this code snippet as is and protect it under an "optimize" flag via the context, since using VectorReader as a component in other parts of the code (if needed) or when this struct is used for unit testing the struct would be complete with all the necessary information.

Reply:
and pass the vectorIndex as part of the context?

Reply:
ah i see, yeah i forgot that the search interface was changed. sounds good to me.